### ISSUE TITLE
[suggestion][docs] "Why Spack" Section in Documentation

### ISSUE THREAD

**Issue by danieldjewell** [LLNL/spack#21976]
### Synopsis
The documentation's [features page](https://spack.readthedocs.io/en/latest/features.html) discusses some differences but doesn't really clearly and succinctly answer the question: "Why Spack?"

### Rationale
<sup>[*In my best [Don LaFontaine](https://en.wikipedia.org/wiki/Don_LaFontaine) voice*]</sup> 
***In a world... where the lines ... between package managers are blurred...***

I think having a clearly defined "Why" also helps to clarify/explain goals/scope... It also helps to ensure that something isn't a solution in search of a problem :grin: :wink:

When I suggest "Why Spack?" It's really more than that: e.g.

* What are the high level goals of Spack?
* What *exactly* makes Spack different (a bulleted list)
  * What are (some/all) of the problems with existing systems? 
  * Concrete examples and comparisons are very helpful here
    *  Example: If true, what makes Spack better than Conda and in what situations?
  * What problems does Spack solve by being different?
* How is Spack designed/designed differently in order to solve the problems above?

And, perhaps most importantly:

***Why should I use Spack? Why should I care?***

(I know this last one is a little ... harshly direct ... but not everyone will immediately understand why. I'm certain that all of the devs understand the answer to this implicitly... but it's extraordinarily important to actually explain it. By analogy, not doing so is somewhat like a Physics professor walking into a Physics 101 class and using vocabulary and concepts way beyond what the students could be expected to know at that point... It happens. It's extraordinarily easy to make assumptions about the level of knowledge of an audience when one is very well versed in the subject matter -- it's rather difficult to step back and question those assumptions. Also, FWIW, simple clear explanations do not imply that one nor the subject matter is unsophisticated ... ) 

### Additional information (tl;dr)

Basically: A tl;dr; of Why. Why Spack exists. Why it's cool. Why it's great. Why it's not so great. Why you'd use it, why you wouldn't. Even some clear real-world use cases/examples. (The aforementioned features page is in the right direction but it's like the details - not the brief summary.)
